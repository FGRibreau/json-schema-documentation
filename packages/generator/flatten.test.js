const Ajv = require('ajv');

test('flatten schema', () => {
  expect.assertions(1);

  const flatten = require('./flatten');

  const schemas = new Ajv({
    // check all rules collecting all errors. Default is to return after the first error.
    allErrors: true,

    // clude the reference to the part of the schema (schema and parentSchema) and validated data in errors (false by default).
    verbose: true,

    // validation of other keywords when $ref is present in the schema
    extendRefs: 'fail',

    //  if the reference cannot be resolved during compilation the exception is thrown
    missingRefs: true,

    // check all rules collecting all errors. Default is to return after the first error.
    allErrors: true,

    // change data type of data to match type keyword
    coerceTypes: false,

    schemas: [require('./fixtures/test.json'), require('./fixtures/defs.json')],
  });

  expect(flatten(schemas.getSchema('FullTest'))).toMatchSnapshot();
});
