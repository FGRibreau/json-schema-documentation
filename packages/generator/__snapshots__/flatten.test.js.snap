// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flatten schema 1`] = `
Object {
  "$id": "FullTest",
  "description": "FullTest-description",
  "properties": Object {
    "configurations": Object {
      "description": "configuration-description",
      "items": Object {
        "properties": Object {
          "endpoint": Object {
            "properties": Object {},
            "required": Array [
              "url",
              "method",
              "pure",
              "parameters",
              "ui",
            ],
            "type": "object",
          },
          "method": Object {
            "enum": Array [
              "GET",
              "POST",
            ],
            "type": "string",
          },
          "version": Object {
            "$ref": "defs.json#/definitions/BasicSemVer",
            "description": "MAJOR.MINOR.PATCH, see https://semver.org/",
            "examples": Array [
              "0.0.1",
              "1.0.0",
            ],
            "pattern": "^[0-9]+\\\\.[0-9]+\\\\.[0-9]+$",
            "title": "Basic Semantic version",
            "type": "string",
          },
        },
        "required": Array [
          "version",
          "endpoint",
        ],
        "type": "object",
      },
      "maxItems": 10,
      "minItems": 1,
      "title": "configuration-object",
      "type": "array",
    },
  },
  "required": Array [
    "configurations",
  ],
  "title": "FullTest-title",
  "type": "object",
}
`;
